crates/sui-core/src/stake_aggregator.rs

```diff
@@ -115,21 +120,62 @@ impl<const STRENGTH: bool> StakeAggregator<AuthoritySignInfo, STRENGTH> {
                     self.data.values().cloned().collect(),
                     self.committee(),
                 ) {
-                    Ok(aggregated) => InsertResult::QuorumReached(aggregated),
+                    Ok(aggregated) => {
+                        match aggregated.verify_secure(
+                            &data,
+                            Intent::default().with_scope(T::SCOPE),
+                            self.committee(),
+                        ) {
+                            // In the happy path, the aggregated signature verifies ok and no need to verify
+                            // individual.
+                            Ok(_) => InsertResult::QuorumReached(aggregated),
+                            Err(_) => {
+                                // If the aggregated signature fails to verify, fallback to iterating through
+                                // all signatures and verify individually. Decrement total votes and continue
+                                // to find new authority for signature to reach the quorum.
+                                //
+                                // TODO(joyqvq): It is possible for the aggregated signature to fail every time
```

Comment:
Yeah it's important to have a follow up PR that maintains the list of those already passed verification.

Another alternative would be to always fallback to individual sigs after the first time the aggregated verification failed (I mean never do aggregated again for this tx - something is already wrong and we should log it for the future)

We should optimize here in an upcoming PR. This PR is in a good state.